File: src/main/java/selector/Selector.java
package selector;

import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.remote.RemoteWebDriver;

class Selector implements SelectorInterface {

    private FirefoxDriver firefoxDriver = null;
    private ChromeDriver chromeDriver = null;

    public boolean elementExists(WebElement element) {
        while (!element.isDisplayed() || !element.isEnabled());
        return true;
    }

    public boolean click(WebElement element) {
        while (!element.isDisplayed() || !element.isEnabled()) {}
        element.click();
        return true;
    }

    public boolean setText(WebElement element, String text) {
        while (!element.isDisplayed() || !element.isEnabled()) {}
        element.sendKeys(text);
        return true;
    }

    public WebElement findElement(WebElement element) {
        if (!elementExists(element)){
            return null;
        }
        while (!element.isDisplayed() || !element.isEnabled()) {}
        return element;
    }

    //----------------------------------------------------------------

    public RemoteWebDriver setFirefoxDriver(FirefoxDriver driver) {
        return this.firefoxDriver = driver;
    }

    public RemoteWebDriver setChromeDriver(ChromeDriver driver) {
        return this.chromeDriver = driver;
    }

    public RemoteWebDriver getDriver() {
        if (firefoxDriver != null) {
            return firefoxDriver;
        }
        return chromeDriver;
    }

}

File: src/main/java/selector/Selector.kt
package selector

import common.Constants.Companion.randomSleepMaxTime
import org.openqa.selenium.WebElement
import org.openqa.selenium.remote.RemoteWebDriver
import selectordriver.SelectorInterface
import java.util.*

abstract class Selector(var driver: RemoteWebDriver) : SelectorInterface {

    override fun elementExists(element: WebElement): Boolean {
        if (element.isDisplayed && element.isEnabled) {
            return true
        }
        return false
    }

    override fun clickLastInList(selectorList: List<WebElement>) {
        click(selectorList[selectorList.size - 1])
    }

    override fun waitForElement(element: WebElement): Boolean {
        while (!element.isDisplayed || !element.isEnabled) {
        }
        return true
    }

    override fun click(element: WebElement): Boolean {
        waitForElement(element)
        element.click()
        return true
    }

    override fun setText(element: WebElement, text: String): Boolean {
        waitForElement(element)
        element.sendKeys(text)
        return true
    }

    override fun openURL(url: String) {
        driver.get(url)
    }

    override fun randomSleep() {
        Thread.sleep(Random().nextInt(randomSleepMaxTime).toLong())
    }

    override fun randomSleep(sleep: Long) {
        Thread.sleep(sleep)
    }


}
